src/main/java/uk/gov/companieshouse/service/dissolution/chips/ChipsResponseService.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.companieshouse.service.dissolution.chips;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.exception.DissolutionNotFoundException;
+import uk.gov.companieshouse.exception.DissolutionVerdictAlreadyExists;
+import uk.gov.companieshouse.mapper.DissolutionVerdictMapper;
+import uk.gov.companieshouse.model.db.dissolution.Dissolution;
+import uk.gov.companieshouse.model.db.dissolution.DissolutionRejectReason;
+import uk.gov.companieshouse.model.db.dissolution.DissolutionVerdict;
+import uk.gov.companieshouse.model.dto.chips.ChipsResponsePostRequest;
+import uk.gov.companieshouse.model.enums.VerdictResult;
+import uk.gov.companieshouse.repository.DissolutionRepository;
+import uk.gov.companieshouse.service.dissolution.DissolutionEmailService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ChipsResponseService {
+
+    private final DissolutionRepository repository;
+
+    private final DissolutionVerdictMapper dissolutionVerdictMapper;
+
+    private final DissolutionEmailService dissolutionEmailService;
+
+    public ChipsResponseService(
+            DissolutionRepository repository,
+            DissolutionVerdictMapper dissolutionVerdictMapper,
+            DissolutionEmailService dissolutionEmailService
+    ) {
+        this.repository = repository;
+        this.dissolutionVerdictMapper = dissolutionVerdictMapper;
+        this.dissolutionEmailService = dissolutionEmailService;
+    }
+
+    public void saveAndNotifyDissolutionApplicationOutcome(ChipsResponsePostRequest body) {
+        final Dissolution dissolution = this.repository
+                .findByDataApplicationReference(body.getSubmissionReference())
+                .orElseThrow(DissolutionNotFoundException::new);
+
+        if (dissolution.getVerdict() != null) {
+            throw new DissolutionVerdictAlreadyExists();
+        }
```

Comment:
This is very different from our other endpoints. Usually we perform these checks at the controller level to separate controller logic from service logic. Usually our services handle happy paths only as validation is done in advance.